api/src/legacy_soap_api/legacy_soap_api_config.py

```diff
@@ -25,3 +32,9 @@ def model_post_init(self, _context: Any) -> None:
             except Exception:
                 # This except is to make sure the API still starts up, even if the value is malformed
                 logger.exception("Could not load soap auth content")
+
+
+@cache
+def get_soap_config() -> LegacySoapAPIConfig:
+    # This is cached since any changes the config will require app restart anyways.
+    return LegacySoapAPIConfig()
```

Comment:
Huh, wasn't aware of `@cache` - probably better than the approach we have been doing using globals to avoid re-making these env var config classes like https://github.com/HHS/simpler-grants-gov/blob/main/api/src/util/file_util.py#L22

Good find.